opencensus/exporters/trace/stackdriver/internal/stackdriver_exporter.cc

```diff
@@ -261,29 +256,52 @@ void Handler::Export(
   ::google::protobuf::Empty response;
   grpc::ClientContext context;
   context.set_deadline(absl::ToChronoTime(absl::Now() + opts_.rpc_deadline));
-  grpc::Status status = stub_->BatchWriteSpans(&context, request, &response);
+  grpc::Status status =
+      opts_.trace_service_stub->BatchWriteSpans(&context, request, &response);
   if (!status.ok()) {
     std::cerr << "BatchWriteSpans failed: "
               << opencensus::common::ToString(status) << "\n";
   }
 }
 
-}  // namespace
-
-// static
-void StackdriverExporter::Register(const StackdriverOptions& opts) {
+std::unique_ptr<google::devtools::cloudtrace::v2::TraceService::Stub>
+MakeStackdriverStub() {
   auto channel = ::grpc::CreateCustomChannel(
       kGoogleStackdriverTraceAddress, ::grpc::GoogleDefaultCredentials(),
       ::opencensus::common::WithUserAgent());
+  return ::google::devtools::cloudtrace::v2::TraceService::NewStub(channel);
+}
+
+}  // namespace
+
+// static
+void StackdriverExporter::Register(StackdriverOptions&& opts) {
+  if (opts.trace_service_stub == nullptr) {
+    opts.trace_service_stub = MakeStackdriverStub();
+  }
+  ::opencensus::trace::exporter::SpanExporter::RegisterHandler(
+      absl::make_unique<Handler>(std::move(opts)));
+}
+
+// static, DEPRECATED
```

Comment:
Does absl provide a deprecated macro?

Reply:
Yep, I use it in the header: https://github.com/census-instrumentation/opencensus-cpp/pull/341/files#diff-a00bc854d48bc0ac9e92a7c77d2f4d0eR51